.github/workflows/release.yml

```diff
@@ -0,0 +1,55 @@
+name: Release
```

Comment:
The Release workflow will run when the Test workflow completes AND is successful AND we are on the main branch. If the git ref is a git tag, create a Github release AND upload package to PyPi. If the git ref is the main branch, upload to Test PyPi.